### PR TITLE
120 implement auto log out ite portal

### PR DESCRIPTION
**Pull request by bbodine1**
- Update environment variables in app.module.ts
- Update user-info routerLink in portal.component.html
- Add APP_INITIALIZER and LastActiveService
- Add OurAuthService to app.module.ts and update appInactiveTimeout in environment files
- Update protectedResourceMap to use Array type instead of string[]
- Update login failed route in MSALGuardConfigFactory
- Fix MSALInstanceFactory function name
- Update MSALInstanceFactory logger options
- auto logout working
- Refactor logout functionality in auth service
- Update environment configurations
- Comment out MSAL logging in app.module.ts

## Summary by CodeRabbit

- **New Features**
	- Enhanced login functionality with the introduction of `loginRedirect` for a smoother user experience.
	- Implemented automatic logout on user inactivity to increase security.

- **Improvements**
	- Streamlined authentication process with Azure Active Directory integration.
	- Improved event handling and error management in the authentication flow.
	- Updated user interface to remove redundant navigation links.

- **Bug Fixes**
	- Fixed issues with account event handling to ensure proper account state across sessions.

- **Configuration**
	- Updated environment configurations to align with new authentication and inactivity timeout settings.

### CODE REVIEW

File: apps/ite-portal/src/app/app.module.ts [hunk 1]
```diff
@@ -48,50 +49,56 @@
...
-    'Making sure correct branch by NX_AD_CLIENT_ID: ' + NX_AD_CLIENT_ID
-  );
-  console.log('Making sure correct branch by NX_AD_TID: ' + NX_AD_TID);
-  console.log('Making sure correct branch by gatewayApiUrl: ' + gatewayApiUrl);
-  console.log('Making sure correct branch by portalApiUrl: ' + portalApiUrl);
+export function MSALInstanceFactory(): IPublicClientApplication {
+  // console.log('Making sure correct branch by NX_AD_CLIENT_ID: ' + clientId);
+  // console.log('Making sure correct branch by NX_AD_TID: ' + tenantId);
+  // console.log('Making sure correct branch by gatewayApiUrl: ' + gatewayApiUrl);
+  // console.log('Making sure correct branch by portalApiUrl: ' + portalApiUrl);
```
[the remark this answers is not in the document]
Comment (bbodine1) [reply apps/ite-portal/src/app/app.module.ts:68]: Leave these for future debugging

File: apps/ite-portal/src/environments/environment.uat.ts [hunk 2]
```diff
@@ -1,9 +1,25 @@
-/* eslint-disable @typescript-eslint/naming-convention */
+const clientId = process.env['NX_AD_CLIENT_ID_PROD'] || '';
+const tenantId = process.env['NX_AD_TID_PROD'] || '';
+const gatewayApiUrl = process.env['NX_GATEWAY_API_PROD'] || '';
```
Comment (kristinmerbach) [inline apps/ite-portal/src/environments/environment.uat.ts:3]: these should be the uat variables not the prod ones

**Review by kristinmerbach — CHANGES_REQUESTED** [review]
Looks great! Only change it looks like you copy and pasted into the uat environment file so need those env variables adjusted

File: apps/ite-portal/src/environments/environment.uat.ts [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
+  NX_GATEWAY_API: gatewayApiUrl,
+  NX_PORTAL_API: portalApiUrl,
+  msalConfig: {
+    auth: {
+      clientId: clientId,
+      authority: authority,
+    },
+  },
+  apiConfig: {
+    uri: authority,
```
[the remark this answers is not in the document]
Comment (bbodine1) [reply apps/ite-portal/src/environments/environment.uat.ts:22]: The current code is correct. Please disregard this suggestion.

**Review by ymhari — APPROVED** [review]

**Pull request merged**